### ISSUE TITLE
Catch-all: XYZ doesn't work with `SpatialData`

### ISSUE THREAD

**Issue by timtreis** [scverse/squidpy#875]
Please just post/link functions here that do not work with `SpatialData` - we will then gradually port those over.

**Comment by timtreis** [comment]
Essentially tried following the [Squidpy Xenium tutorial](https://squidpy.readthedocs.io/en/stable/notebooks/tutorials/tutorial_xenium.html) but using a `spatialdata` object. Failed when I tried to color my plot by Leiden colors.

**Comment by giovp** [comment]
> Essentially tried following the [Squidpy Xenium tutorial](https://squidpy.readthedocs.io/en/stable/notebooks/tutorials/tutorial_xenium.html) but using a `spatialdata` object. Failed when I tried to color my plot by Leiden colors.

with this?

https://github.com/scverse/squidpy_notebooks/pull/125

**Comment by timtreis** [comment]
No, this one seems to basically just extract the `adata` and then does normal things that were possible before. I think the goal should be to directly pass either the `sdata.tables[...]` or `sdata` and `table_name` (if the latter cannot be inferred), wdyt? 🤔

**Comment by LLehner** [comment]
> No, this one seems to basically just extract the `adata` and then does normal things that were possible before. I think the goal should be to directly pass either the `sdata.tables[...]` or `sdata` and `table_name` (if the latter cannot be inferred), wdyt? 🤔

We did that initially but to keep verbosity low, picked the adata approach.
